### PR TITLE
Kcache epflush

### PR DESCRIPTION
**Pull request by cosunae**

### CODE REVIEW

File: src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp [hunk 1]
```diff
@@ -34,14 +35,90 @@ std::string CodeGeneratorHelper::indexIteratorName(Array3i dims) {
...
+    const IndexRange<const std::unordered_map<int, iir::Field>>& tempFields,
+    const std::shared_ptr<iir::StencilInstantiation>& stencilInstantiation,
+    const std::unique_ptr<iir::MultiStage>& ms, CodeGeneratorHelper::FunctionArgType funArg) {
+
+  std::unordered_set<std::string> processedDims;
+  std::vector<std::string> strides;
+  for(auto field : nonTempFields) {
+    const auto fieldName = stencilInstantiation->getNameFromAccessID((*field).second.getAccessID());
+    Array3i dims{-1, -1, -1};
+    // TODO this is a hack, we need to have dimensions also at ms level
```
Comment (twicki) [inline src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp:64]: why? If we need it once, we can fetch it there. It is always a trade off between duplication the information (and keeping it up to date everywhere) and accessing it from somewhere else, right?

File: src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp [hunk 2]
```diff
@@ -34,14 +35,90 @@ std::string CodeGeneratorHelper::indexIteratorName(Array3i dims) {
...
+}
+
+iir::Extents CodeGeneratorHelper::computeTempMaxWriteExtent(iir::Stencil const& stencil) {
+  auto tempFields = makeRange(
+      stencil.getFields(),
+      std::function<bool(std::pair<int, iir::Stencil::FieldInfo> const&)>(
+          [](std::pair<int, iir::Stencil::FieldInfo> const& p) { return p.second.IsTemporary; }));
+  iir::Extents maxExtents{0, 0, 0, 0, 0, 0};
+  for(auto field : tempFields) {
+    DAWN_ASSERT((*field).second.field.getWriteExtentsRB().is_initialized());
```
Comment (twicki) [inline src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp:116]: why deref field here?

File: src/dawn/CodeGen/Cuda/CodeGeneratorHelper.h [hunk 3]
```diff
@@ -26,6 +27,8 @@ namespace cuda {
 
 class CodeGeneratorHelper {
 public:
+  enum class FunctionArgType { caller, callee };
```
Comment (twicki) [inline src/dawn/CodeGen/Cuda/CodeGeneratorHelper.h]:
```suggestion
  enum class FunctionArgType {FT_Caller, FT_Callee };
```

**Review by twicki — CHANGES_REQUESTED** [review]
I assume the functionality moved to MSGenerator did not change? I did not review this part

File: src/dawn/CodeGen/Cuda/CodeGeneratorHelper.h (hunk 3 above)
Comment (cosunae) [reply src/dawn/CodeGen/Cuda/CodeGeneratorHelper.h]: done

File: src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp (hunk 2 above)
Comment (cosunae) [reply src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp:116]: clarified

File: src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp (hunk 1 above)
Comment (cosunae) [reply src/dawn/CodeGen/Cuda/CodeGeneratorHelper.cpp:64]: clarified

**Pull request closed without merging**